ocdskingfisherprocess/database.py

```diff
@@ -758,17 +758,25 @@ def _delete_orphan_data_data(self):
                 )
 
     def _delete_orphan_data_package_data(self):
-        data = {}
-        sql = """DELETE FROM package_data
-                WHERE id NOT IN
-                    (
-                        SELECT package_data_id FROM release union
-                        SELECT package_data_id FROM record
-                    );"""
+        data_get = {}
+        sql_get = """SELECT package_data.id FROM package_data
+                       LEFT JOIN release ON release.package_data_id = package_data.id
+                       LEFT JOIN record ON record.package_data_id = package_data.id
+                       WHERE release.package_data_id IS NULL AND record.package_data_id IS NULL
+                       LIMIT 10000;"""
         logger = logging.getLogger('ocdskingfisher.database.delete-collection')
         logger.debug("Deleting package_data")
-        with self.get_engine().begin() as connection:
-            connection.execute(sa.sql.expression.text(sql), data)
+        while True:
```

Comment:
Instead of doing a fresh query to get 10,000 at a time, can't we use a [server-side cursor](https://www.psycopg.org/docs/usage.html#server-side-cursors) to perform a single SELECT, use it to `fetchmany(10000)` at a time, and then delete those?

Reply:
We could look at that, but on the other hand we don't fully know how that would perform on a loaded server - maybe the SELECT would take a very long time? But we know the "LIMIT" method, while crude, does work because it's what we already use for the data table. 

Let me know if you want me to look at this or not.

Reply:
Let's instead create an issue about considering server-side cursors where we use large LIMITs.

Reply:
Issue made